packages/analytics/amplify_analytics_pinpoint/example/integration_test/identify_user_test.dart

```diff
@@ -0,0 +1,144 @@
+
+    setUpAll(() async {
+      eventsStream = await configureAnalytics();
+    });
+
+    tearDownAll(Amplify.reset);
+
+    // TODO(fjnoyp): Running this locally too many times will cause max endpoint error of 15 from Pinpoint
+    // We need to delete/reset Analytics backend somehow
+    test(
+      'properties of identityUser() added to all future events',
+      () async {
+        const userId = 'userId';
+        const name = 'name';
+        const email = 'email';
+        const plan = 'plan';
+
+        const latitude = 43.0;
+        const longitude = 12.0;
+        const postalCode = '94070';
+        const city = 'city';
+        const region = 'region';
+        const country = 'USA';
+
+        final location = AnalyticsUserProfileLocation(
+          latitude: latitude,
+          longitude: longitude,
+          postalCode: postalCode,
+          city: city,
+          region: region,
+          country: country,
+        );
+
+        final properties = AnalyticsProperties()
+          ..addBoolProperty(boolProperty.key, boolProperty.value)
+          ..addDoubleProperty(doubleProperty.key, doubleProperty.value)
+          ..addIntProperty(intProperty.key, intProperty.value)
+          ..addStringProperty(stringProperty.key, stringProperty.value);
+
+        try {
+          await Amplify.Analytics.identifyUser(
+              userId: userId,
+              userProfile: AnalyticsUserProfile(
+                name: name,
+                email: email,
+                plan: plan,
+                location: location,
+                analyticsProperties: properties,
+              ));
+
+          const customEventName = 'my custom event type name';
+          final customEvent = AnalyticsEvent(customEventName);
+
+          expect(
+            eventsStream,
+            emits(allOf([
+              containsPair('event_type', customEventName),
+              // Session is valid before timestamp
+              containsPair(
+                  'endpoint',
+                  allOf(
+                    containsPair('EndpointStatus', 'ACTIVE'),
+                    containsPair('OptOut', 'ALL'),
+                    containsPair(
+                      'Location',
+                      allOf(
+                        containsPair('Latitude', latitude),
+                        containsPair('Longitude', longitude),
+                        containsPair('PostalCode', postalCode),
+                        containsPair('City', city),
+                        containsPair('Region', region),
+                        containsPair('Country', country),
+                      ),
+                    ),
+                    containsPair(
+                      'User',
+                      containsPair(
+                        'UserId',
+                        userId,
+                      ),
+                    ),
+                    // TODO(fjnoyp): Don't know why this is failing!
+                    // expected:
```

Comment:
Are strings being JSON-encoded somewhere?

Reply:
Resolved it now, yeah it's because of that and I needed to use your stringifiedboolproperty!